scarpe-components/lib/scarpe/components/calzini/art_widgets.rb

```diff
@@ -36,10 +36,20 @@ def rect_element(props)
   end
 
   def line_element(props)
+    dc = props["draw_context"] || {}
+    cap = dc["cap"]
+    if cap == :curve
+      cap = "round"
+    elsif cap == :rect
+      cap = "butt"
+    elsif cap == :project
+      cap = "square"
+    end
+
     HTML.render do |h|
       h.div(id: html_id, style: line_div_style(props)) do
         h.svg(width: props["x2"], height: props["y2"]) do
-          h.line(x1: props["left"], y1: props["top"], x2: props["x2"], y2: props["y2"], style: line_svg_style(props))
+          h.line(x1: props["left"], y1: props["top"], x2: props["x2"], y2: props["y2"], style: line_svg_style(props), "stroke-linecap":cap,"transform-origin": "center right")
```

Comment:
You can add these using symbols with the colon-string syntax:

    :"stroke-linecap": cap, :"transform-origin": "center right"

That will then not set stroke-linecap any time cap is nil, for instance, as well as matching the rest of the properties everywhere else.

Reply:
oo okay